language-server/modules/langserver-core/src/main/java/org/ballerinalang/langserver/util/definition/LSStandardLibCache.java

```diff
@@ -0,0 +1,233 @@
+    }
+
+    /**
+     * Get the standard library cache root which resides within the temp directory.
+     * 
+     * @return {@link Path} Cache root
+     * @throws LSStdlibCacheException If the cache root initialization has been failed
+     */
+    public Path getStdlibCacheRoot() throws LSStdlibCacheException {
+        if (!cacheProjectRootSuccess) {
+            throw new LSStdlibCacheException("Cache Root could not initialized");
+        }
+        return stdlibCacheProjectPath;
+    }
+
+    private List<TopLevelNode> getNodesForModule(String moduleName) throws UnsupportedEncodingException {
+        Compiler compiler = getCompiler(stdlibCacheProjectPath.toString());
+        BLangPackage bLangPackage = compiler.compile(moduleName);
+        return bLangPackage.topLevelNodes.stream()
+                .filter(topLevelNode -> {
+                    Set<Flag> flagSet;
+                    switch (topLevelNode.getKind()) {
+                        case FUNCTION:
+                            flagSet = ((BLangFunction) topLevelNode).flagSet;
+                            break;
+                        case TYPE_DEFINITION:
+                            flagSet = ((BLangTypeDefinition) topLevelNode).flagSet;
+                            break;
+                        case CONSTANT:
+                            flagSet = ((BLangConstant) topLevelNode).flagSet;
+                            break;
+                        // TODO: Handle XML Namespace Declarations
+                        case ANNOTATION:
+                            flagSet = ((BLangAnnotation) topLevelNode).flagSet;
+                            break;
+                        default:
+                            flagSet = new HashSet<>();
+                            break;
+                    }
+                    return flagSet.contains(Flag.PUBLIC);
+                })
+                .collect(Collectors.toList());
+    }
+
+    private CompilerContext createNewCompilerContext(String projectDir) {
+        CompilerContext context = new CompilerContext();
+        CompilerOptions options = CompilerOptions.getInstance(context);
+        options.put(PROJECT_DIR, projectDir);
+        options.put(COMPILER_PHASE, CompilerPhase.DESUGAR.toString());
+        options.put(PRESERVE_WHITESPACE, Boolean.toString(false));
+        options.put(OFFLINE, Boolean.toString(true));
+        options.put(CompilerOptionName.EXPERIMENTAL_FEATURES_ENABLED, Boolean.toString(true));
+        context.put(SourceDirectory.class, new FileSystemProjectDirectory(Paths.get(projectDir)));
+        context.put(DiagnosticListener.class, new DefaultDiagnosticListener());
+        return context;
+    }
+
+    private Compiler getCompiler(String projectDir) throws UnsupportedEncodingException {
+        Compiler compiler = Compiler.getInstance(createNewCompilerContext(projectDir));
+        compiler.setOutStream(new LSCompilerUtil.EmptyPrintStream());
+
+        return compiler;
```

Comment:
Shall we add a traceLog for this compilation. Will be useful when analyzing performance bugs. https://github.com/ballerina-platform/ballerina-lang/blob/master/language-server/modules/langserver-compiler/src/main/java/org/ballerinalang/langserver/compiler/LSModuleCompiler.java#L287-L292